src/main/java/io/anserini/index/generator/GeoGenerator.java

```diff
@@ -42,17 +42,39 @@ public Document createDocument(JsonCollection.Document geoDoc) {
 
           Field[] fields = new Field[0];
           if (shape instanceof Line) {
-            fields = LatLonShape.createIndexableFields("geometry", (Line) shape);
+            Line line = (Line) shape;
+            fields = LatLonShape.createIndexableFields("geometry", line);
+
+            for (int i = 0; i < line.numPoints(); ++i) {
```

Comment:
Can you change to `i++`.
Also, consider if a functional style using `forEach()` will make the code look cleaner?

Reply:
Changed to `i++`.

Perhaps `forEach()` would be hard to use in this case, since we are iterating over two arrays (`getLats()` and `getLons()`)?